tests/flytekit/unit/core/test_type_hints.py

```diff
@@ -2000,3 +2007,59 @@ def my_wf(a: int, retries: int) -> int:
 
     with pytest.raises(AssertionError):
         my_wf(a=1, retries=1)
+
+
+def test_unsafe_input_wf_and_task():
+    @task(unsafe=True)
+    def t1(a) -> int:
+        if type(a) == int:
+            return a + 1
+        return 0
+
+    @task
+    def t2_wo_unsafe(a) -> int:
+        return a + 1
+
+    @workflow(unsafe=True)
+    def wf1_with_unsafe(a) -> int:
+        return t1(a=a)
+
+    assert wf1_with_unsafe(a=1) == 2
+    assert wf1_with_unsafe(a="1") == 0
+    assert wf1_with_unsafe(a=None) == 0
+
+    @workflow
+    def wf1_wo_unsafe(a) -> int:
+        return t1(a=a)
+
+    @workflow
+    def wf1_wo_unsafe2(a: int) -> int:
```

Comment:
Did you test this workflow in the sandbox cluster?

Reply:
No, I only test it on my local environment. Can I ask how to try it?

Reply:
This pr can help you, you can either use imageSpec or Dockerfile to create your own image.
https://github.com/flyteorg/flytekit/pull/1870

Reply:
```
from flytekit import task, Secret, workflow, ImageSpec

flytekit_dev_version = "https://github.com/Mecoli1219/flytekit.git@c7c9a2ad2bdf6799ec324955a023281cca030038"
image = ImageSpec(
    registry="localhost:30000",
    platform="linux/amd64",
    apt_packages=["git"],
    packages=[
        f"git+{flytekit_dev_version}",
    ],
)


@task(unsafe=True, container_image=image)
def t1(a) -> int:
    if type(a) == int:
        return a + 1
    return 0


@workflow(unsafe=True)
def wf1_with_unsafe(a) -> int:
    return t1(a=a)
```

Reply:
The above code will fail. The reason is detailed in https://github.com/flyteorg/flyte/pull/5261.